File: SanityEngine.Editor/Editor/SanityEditor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Sanity.Concurrency;

using Windows.Foundation;

namespace Sanity.Editor
{
    class SanityEditor : IDisposable
    {
        public SanityEngine Engine
        {
            get; private set;
        }

        private WinRTNursery nursery = new();

        public SanityEditor()
        {
            nursery.StartSoon(RunUi);
        }

        public void Tick(bool windowVisible)
        {
            

            Engine.Tick(windowVisible);
        }

        public void SetWindowSize(Size size)
        {
        }

        public void Dispose() => ((IDisposable)nursery).Dispose();

        /// <summary>
        /// Does the needful to keep the UI chugging along. Runs in a separate thread. Yay.
        /// </summary>
        private void RunUi()
        {
        }

    }
}

File: SanityEngine/Editor/SanityEditor.cpp
#include "SanityEditor.hpp"

namespace Sanity::Editor {
    SanityEditor::SanityEditor() {
        auto* window = glfwCreateWindow(640, 480, "Sanity Engine", nullptr, nullptr);
    	
        Settings settings{};

        sanity_engine = std::make_unique<SanityEngine>(settings);
    }
} // namespace Sanity::Editor
